applications/conceptual-model-editor/src/app/diagram/reactflow/components/menu-options.tsx

```diff
@@ -43,40 +45,74 @@ export const useMenuOptions = () => {
         deleteHandler?: () => void;
         position?: string;
     }) => {
-        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler } = props;
+        const { openDetailHandler, createProfileHandler, modifyHandler, removeFromViewHandler, deleteHandler, position } = props;
+        // You have to actually define the type here explictly, typescript can't interfere the type correctly
```

Comment:
One more reason to have a better structure.
We may even have a builder like:
```js
createOptionsMenuBuilder()
  .addAction("Detail", openDetailHandler)
  .addAction("Modify", modigyHandler)
  .addSeparator()
  .build().
```
What do you think?